go.mod

```diff
@@ -19,8 +19,8 @@ require (
 	github.com/elastic/elastic-agent-client/v7 v7.0.0-20210407144825-cc1c33cfa1d0 // indirect
 	github.com/elastic/elastic-package v0.0.0-20210310173719-3b8f76516ae3
 	github.com/elastic/gmux v0.1.0
-	github.com/elastic/go-elasticsearch/v7 v7.11.0
-	github.com/elastic/go-elasticsearch/v8 v8.0.0-20201007143536-4b4020669208
+	github.com/elastic/go-elasticsearch/v7 v7.13.1
```

Comment:
sorry I was a bit unclear earlier, we should be using the 7.14 branch of go-elasticsearch

`go get github.com/elastic/go-elasticsearch/v7@7.14` should do it

Reply:
I'm caught in some weird dependency loop, I think because go is seeing `7.14` as being tagged with version `v7.5`:

```
# go get github.com/elastic/go-elasticsearch/v7@7.14
go get: removed github.com/elastic/elastic-package v0.0.0-20210310173719-3b8f76516ae3
go get: downgraded github.com/elastic/go-elasticsearch/v7 v7.9.0 => v7.5.1-0.20210701130713-3c08ee738285

# make update
// elided ..
no required module provides package github.com/elastic/elastic-package; to add it:
        go get github.com/elastic/elastic-package

# go get github.com/elastic/elastic-package@3b8f76516ae3
go get: added github.com/elastic/elastic-package v0.0.0-20210310173719-3b8f76516ae3
go get: upgraded github.com/elastic/go-elasticsearch/v7 v7.5.1-0.20210701130713-3c08ee738285 => v7.9.0
```

When I try to upgrade to the 7.14 branch, it removes `elastic-package`, but when I re-add it, go mod "upgrades" to `v7.9.0`..

@Anaethelion do you have any ideas?

Reply:
There's a version conflict: `elastic-package` requires [go-elasticsearch `v.7.9.0`](https://github.com/elastic/elastic-package/blob/master/go.mod#L10); but the `go-elasticsearch 7.14` branch is resolved into `v7.5.1` (with the latest commit, but still the wrong version). The dependency has been resolved similar in the past, but it becomes an issue now that `elastic-package` is a dependency. 

Here comes my wild guess: 
Up until `7.5` the ES client releases were tagged from the `7.x` branch, starting with `7.6` dedicated version branches are created. This means that up until the `7.14` release, this branch (created from `7.x`) is still associated with the `7.5.x` tag. Does that sound reasonable?

Are there any major changes in the `7.14` branch that we need to get into the APM Server for the features that we are using? If not - I suggest to update to the released `7.13` branch instead.

Reply:
Another couple of options:
 - We could move the elastic-package dep to a separate go.mod; I ended up doing this for the Go Agent: https://github.com/elastic/apm-agent-go/blob/master/scripts/jenkins/jenkins.go.mod 
 - I _think_ we could use a `replace` directive

Reply:
I'm sort of inclined to use officially tagged versions and use `v7.13.1` for now and keep it simple

Reply:
@simitt is right, for reference this is the `go mod` doc for [version](https://golang.org/ref/mod#versions), [version resolution strategies](https://golang.org/ref/mod#minimal-version-selection) and a simpler explanation on [jfrog's blog](https://jfrog.com/blog/go-big-with-pseudo-versions-and-gocenter/).

For the future I can offer to create a `v7.x.0-pre` tag upon each branch creation to ease understanding before the release tag. I'll check for incompatibility and get back to the team if it's doable!